### ISSUE TITLE
Error embedding font in PDF

### ISSUE THREAD

**Issue by kryops** [kekee000/fonteditor-core#58]
Hi there,

I have discovered an obscure issue when using a font generated with [fonteditor-core](https://github.com/kekee000/fonteditor-core)  in a PDF generated with [PDFKit](https://github.com/foliojs/pdfkit). I wasn't able to find out where the problem actually comes from, so I'm opening this issue in both repos 😅 

When  embedding the generated font into a PDF file, something seems to be off - most PDF viewers are fine, but Adobe Reader displays the following message:

![image](https://github.com/kekee000/fonteditor-core/assets/1042594/1c916530-29cb-4f28-8f32-9926c790684f)

I found out that https://github.com/kekee000/fonteditor-core/commit/78a19e34fa6111192384080f5e18ce88e5d92ac2 is the commit that causes the problem - I'm just not sure whether the generated font is somehow invalid, or PDFKit has a problem embedding the whitespace glyfs that were added in this commit.

I made a minimal reproduction in https://github.com/kryops/fonteditor-pdfkit-bug - in the `dist` folder there are the font files and PDFs generated with different versions of `fonteditor-core`.

Could you have a look into it? Thanks so much!

**Comment by kekee000** [comment]
> @kekee000 thanks for having a look!
> 
> I tried v2.2.2 and updated my reproduction repo, but the error is still there.
> 
> I also added tried to add logging for any differences in the generated font, this is the result when comparing the output of 2.2.2 and 2.1.2 to the (working) one in 2.1.1:
> 
> ```
> >>>  fontEditor-current (2.2.2)
> [diff to 2.1.1] OS/2.xAvgCharWidth: 1056 => 1053
> [diff to 2.1.1] head.checkSumAdjustment: 3988040697 => 4172262679
> [diff to 2.1.1] maxp.maxComponentElements: 1421 => 6
> 
> >>>  fontEditor-2.1.2
> [diff to 2.1.1] OS/2.xAvgCharWidth: 1056 => 1053
> [diff to 2.1.1] hhea.minRightSideBearing: -1034 => 0
> [diff to 2.1.1] hhea.xMaxExtent: 4188 => 0
> [diff to 2.1.1] head.checkSumAdjustment: 3988040697 => 240887043
> ```

according to apple develop doc, remove empy glyph data will be better, it will be fixed at v2.3.0.

```
 If a glyph has zero contours, it need not have any glyph data. If the glyph is made up of components,
 it is recommended that the value -1 be used in this field.
```

https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6glyf.html

**Comment by kryops** [comment]
I can confirm that v2.3.0 fixes my issue. Thanks so much for the quick fix!
